### ISSUE TITLE
Mapping modal <-> form that isn't 1:1

### ISSUE THREAD

**Issue by brianlow** [jakub-zawislak/formex#8]
Can the modal and form use different field names / different structure?

```
defmodule App.Article do
  defstruct [:name, :properties] # name vs title, :properties is a map with :content key
end
```

```
defmodule App.ArticleType do
  use Formex.Type

  def build_form(form) do
    form
    |> add(:title, :text_input)
    |> add(:content, :textarea)
  end
end
```

**Comment by jakub-zawislak** [comment]
Unfortunately no. I didn't even thought about it 😄  I think that it should be easy to make such functionality

**Comment by brianlow** [comment]
I guess you would transform Article into another intermediate struct that matched the shape of the form. Thanks!

**Comment by jakub-zawislak** [comment]
I have just released a 0.5.5 version of formex. Use the new `struct_name` option:

```elixir
  def build_form(form) do
    form
    |> add(:title, :text_input, struct_name: :name)
    |> add(:content, :textarea, struct_name: :properties)
  end
```

If you are using formex_ecto package, you have to update it too
